### ISSUE TITLE
Doesn't work with angular 1.6.1

### ISSUE THREAD

**Issue by vigyanhoon** [DlhSoftTeam/Angular-Kanban#3]
**1) We get following error on angular 1.6.1**

TypeError: Cannot read property 'length' of undefined
    at new controller (DlhSoft.Kanban.Angular.Components.js:131)
    at Object.invoke (angular.js:4842)
    at $controllerInit (angular.js:10695)
    at nodeLinkFn (angular.js:9572)
    at angular.js:9980
    at processQueue (angular.js:16648)
    at angular.js:16692
    at Scope.$eval (angular.js:17972)
    at Scope.$digest (angular.js:17786)
    at Scope.$apply (angular.js:18080)

Its because item is undefined on line 131 of DlhSoft.Kanban.Angular.Components.js

**2) Also it threw following error which could be solved by setting restrict as A (line 75), replace to false (line 76) and using attribute in html:**

angular.js:14328 Error: [$compile:tplrt] Template for directive 'dsKanbanBoard' must have exactly one root element. DlhSoft.Kanban.Angular.Components.Templates/kanban-board.html
http://errors.angularjs.org/1.6.1/$compile/tplrt?p0=dsKanbanBoard&p1=DlhSoft.Kanban.Angular.Components.Templates%2Fkanban-board.html
    at angular.js:68
    at angular.js:9920
    at processQueue (angular.js:16648)
    at angular.js:16692
    at Scope.$eval (angular.js:17972)
    at Scope.$digest (angular.js:17786)
    at Scope.$apply (angular.js:18080)
    at done (angular.js:12210)
    at completeRequest (angular.js:12436)
    at XMLHttpRequest.requestLoaded (angular.js:12364)

**Html:**
<div ds-kanban-board ... </div>

Works perfectly with angular 1.4.9 but doesn't work with 1.6.1 throwing error 1.

**Comment by DlhSoftTeam** [comment]
1) We did reproduce first issue, which seems to do with different behavior of Angular 1.6.1 regarding bindToController. To fix it we introduced a section of code to early bind scope fields to controller fields in case they are not already set by Angular by the time we need this operation completed. The fix is in the updated release 1.0.1.

2) We cannot reproduce second issue. Note that the template in DlhSoft.Kanban.Angular.Components.Templates/kanban-board.html from our original code does have a single root div element, so the error message is either misleading or it indicates that you changed that template in your app only. However, thanks for sharing the workaround you found for your situation!
